src/python/pants/init/plugin_resolver.py

```diff
@@ -91,28 +106,81 @@ def _resolve_plugins(self):
+        #       requests-2.23.0-py2.py3-none-any.whl-install/
+        #
+        #    The plugins-<hash>.txt file that records plugin locations contained the un-installed
+        #    wheel file path:
+        #
+        #    $ cat ~/.cache/pants/plugins/plugins-418c36b574edbcf4720b266b0709750ad588c281.txt
+        #    /home/jsirois/.cache/pants/plugins/requests-2.23.0-py2.py3-none-any.whl
+        #
+        # 2. Now Pex resolves an installed wheel chroot directory and we copy that directory to the
+        #    "-install" directory:
+        #
+        #    ~/.cache/pants/plugins/
+        #      installed_wheels/6ce6cd759a2d13badb1f6b9e665e2aded7a012dd/requests-2.23.0-py2.py3-none-any.whl/
+        #      requests-2.23.0-py2.py3-none-any.whl-install/
+        #
+        #    The plugins-<hash>.txt file that records plugin locations now contains the final
+        #    installed wheel path with the "-install" suffix omitted which leads to the same file
+        #    contents as past Pants versions:
+        #
+        #    $ cat ~/.cache/pants/plugins/plugins-418c36b574edbcf4720b266b0709750ad588c281.txt
+        #        /home/jsirois/.cache/pants/plugins/requests-2.23.0-py2.py3-none-any.whl
+        #
+        #    We add the suffix on after reading the file to find the actual installed wheel path.
+
+        wheel_basename = os.path.basename(distribution.location)
+        plugin_path = os.path.join(self.plugin_cache_dir, wheel_basename)
+
+        with temporary_dir() as td:
+            temp_install_dir = os.path.join(td, wheel_basename)
+            shutil.copytree(distribution.location, temp_install_dir)
+            safe_concurrent_rename(temp_install_dir, self._plugin_location(plugin_path))
+            return plugin_path
 
     @property
-    def plugin_cache_dir(self):
+    def plugin_cache_dir(self) -> str:
         """The path of the directory pants plugins bdists are cached in."""
         return self._plugin_cache_dir
 
     @memoized_property
-    def _python_repos(self):
+    def _python_repos(self) -> PythonRepos:
         return self._create_global_subsystem(PythonRepos)
 
-    def _create_global_subsystem(self, subsystem_type):
-        options_scope = subsystem_type.options_scope
+    def _create_global_subsystem(self, subsystem_type: Type[S]) -> S:
+        options_scope = cast(str, subsystem_type.options_scope)
 
         # NB: The PluginResolver runs very early in the pants startup sequence before the standard
         # Subsystem facility is wired up.  As a result PluginResolver is not itself a Subsystem with
         # PythonRepos as a dependency.  Instead it does the minimum possible work to hand-roll
         # bootstrapping of the Subsystems it needs.
-        known_scope_infos = [
-            ksi
-            for optionable in [GlobalOptionsRegistrar, PythonRepos]
-            for ksi in optionable.known_scope_infos()
+        optionables: Tuple[Type[Optionable], ...] = (GlobalOptionsRegistrar, PythonRepos)
```

Comment:
If this is always this hardcoded pair, technically `Tuple[Type[Optionable], Type[Optionable]]` is more precise. The current hint is accurate, though, and fine to keep.

Reply:
It's more precise but does not express usage. Usage doesn't care how many slots the tuple has. Fixed for real.